samples/java-application-with-reflection/settings.gradle

```diff
@@ -39,4 +39,12 @@
  * SOFTWARE.
  */
 
+pluginManagement {
+    repositories {
+        mavenLocal()
```

Comment:
:x: Shouldn't use `mavenLocal`

Reply:
I found it useful when I have published the gradle plugin itself to the `mavenLocal` (the same way I would use those samples with maven plugin). I moved `mavenLocal` under the `mavenCentral`. Is there some specific reason why we would want to avoid this?

Reply:
There are many reasons to avoid `mavenLocal()`, main one being non reproducibility. There are 2 cases we want to handle:

1. a user building this project, in which case the tests will automatically inject the required artifacts via composite builds, so `mavenLocal()` isn't required
2. a user willing to try a sample, in which case they should use a release (or pre-release) version, in which case `mavenLocal()` is also redundant, and worse, could introduce non-determinism

IMO it's a bad practice to encourage using `mavenLocal()` as people tend to copy and paste samples, and that `mavenLocal()` would be a bad thing to do. Note that this change is also unrelated to this PR, so I would leave it out.

Reply:
I removed `mavenLocal` from this PR so we should be OK.